app/models/host_facet_base.rb

```diff
@@ -0,0 +1,43 @@
+class HostFacetBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  belongs_to_host
+
+  # Add facet's details to host's ENC by returning it from this method
+  # Basic ENC hash structure:
+  # ---
+  # classes:
+  #   ...
+  # parameters:
+  #   ...
+  # environment:
+  #   ...
+  def info
+    {}
+  end
+
+  # Specify any smart proxy id's used by this facet.
+  def smart_proxy_ids
```

Comment:
Why smart proxy *id's* vs smart proxies (objects)? The latter option seems easier to use.